lib/material_datepicker/material_datepicker.dart

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:html';
+
+import 'package:angular/angular.dart';
+import 'package:intl/intl.dart';
+import 'package:quiver/time.dart';
+import 'package:angular_components/button_decorator/button_decorator.dart';
+import 'package:angular_components/content/deferred_content.dart';
+import 'package:angular_components/focus/focus.dart';
+import 'package:angular_components/focus/keyboard_only_focus_indicator.dart';
+import 'package:angular_components/glyph/glyph.dart';
+import 'package:angular_components/laminate/enums/alignment.dart';
+import 'package:angular_components/laminate/popup/popup.dart';
+import 'package:angular_components/material_datepicker/calendar.dart';
+import 'package:angular_components/material_datepicker/date_input.dart';
+import 'package:angular_components/material_datepicker/material_calendar_picker.dart';
+import 'package:angular_components/material_datepicker/module.dart';
+import 'package:angular_components/material_datepicker/range.dart';
+import 'package:angular_components/material_input/material_input.dart';
+import 'package:angular_components/material_list/material_list_item.dart';
+import 'package:angular_components/material_popup/material_popup.dart';
+import 'package:angular_components/material_select/dropdown_button.dart';
+import 'package:angular_components/material_select/material_select_item.dart';
+import 'package:angular_components/mixins/focusable_mixin.dart';
+import 'package:angular_components/model/date/date.dart';
+import 'package:angular_components/utils/angular/properties/properties.dart';
+
+/// A material-design-styled single date picker -- a date parsing input and
+/// calendar picker. Users can type in their own custom dates, or click on the
+/// calendar to select a date.
+///
+/// When a user types in a date, dates with 2-digit years are handled specially.
+/// E.g. 7/7/77 is interpreted as July 7, 1977, not July 7, 77. This logic looks
+/// 20 years into the future: right now (August 2015), "35" is interpreted as
+/// 2035 but "36" is interpreted as "1936". Next year, "36" will start to be
+/// interpreted as 2036.
+///
+/// See also `material-date-range-picker` if you want to choose date ranges.
+///
+/// This component is based on the spec at
+/// <http://folio/ritaho/Ads%20UX%20components/datepicker2014/Latest>.
```

Comment:
This URL (http://folio/ritaho/Ads%20UX%20components/datepicker2014/Latest) doesn't work for me. (And the `<` and `>` can be dropped.)

Note that it is used in a few places.

Reply:
Yeah, that looks like it is an internal only link. I can't find a public version so I'll remove it.